### PR TITLE
WAZO-2176 pre-dial handlers: add doc for pre-dial handlers

### PR DESCRIPTION
**Pull request by pc-m**

### CODE REVIEW

File: content/uc-doc/api_sdk/subroutine.md [hunk 1]
```diff
@@ -106,6 +106,32 @@ You can also use a global subroutine for call forward.
     XIVO_PRESUBR_FWD_CUSTOM = xivo-subrfwd-custom
     XIVO_PRESUBR_FWD_EXTENSION = xivo-subrfwd-extension
 
+Pre-dial handlers
+=================
+
+A pre-dial handler is a subroutine that is executed on a channel before it enters it starts ringing.
```
Comment (sduthil) [inline content/uc-doc/api_sdk/subroutine.md:112]:
```suggestion
A pre-dial handler is a subroutine that is executed on the other side of a channel before it starts ringing:  subroutine allows modifying the channel of the caller party, while a pre-dial handler allows modifying the channel of the called party, just before the called party starts to ring.
```

File: content/uc-doc/api_sdk/subroutine.md [hunk 2]
```diff
@@ -106,6 +106,32 @@ You can also use a global subroutine for call forward.
...
+  * setting a variable on an outgoing channel that should not be set on the current channel.
+  * record an outgoing channel
+
+Here's an example of a pre-dial handler which adds a SIP header on a channel
+
+    [add-sip-header-outgoing]
+    exten = s,1,Set(PJSIP_HEADER(add,X-MY-HEADER)=123)
+    same = n,Return()
+
+To add this pre-dial handler to an outgoing call create a subroutine
```
Comment (sduthil) [inline content/uc-doc/api_sdk/subroutine.md:126]:
```suggestion
To add this pre-dial handler to an outgoing call, create a subroutine:
```

File: content/uc-doc/api_sdk/subroutine.md [hunk 3]
```diff
@@ -106,6 +106,32 @@ You can also use a global subroutine for call forward.
...
+
+    [add-sip-header-pre-dial]
+    exten = s,1,GoSub(wazo-add-pre-dial-hook,s,1(add-sip-header-outgoing))
+    same = n,Return()
+
+Finally, you can set the subroutine of you outgoing call to `add-sip-header-pre-dial`.
```
Comment (sduthil) [inline content/uc-doc/api_sdk/subroutine.md:132]:
```suggestion
Finally, you can set the subroutine of your outgoing call to `add-sip-header-pre-dial`.
```

**Review by sduthil — CHANGES_REQUESTED** [review]

**Comment by sduthil** [comment]
@pc-m I don't see the suggestions included in your new commit, was that on purpose?

**Comment by pc-m** [comment]
> @pc-m I don't see the suggestions included in your new commit, was that on purpose?

No, it's not. I made the changes yesterday messed my rebase this morning

**Comment by pc-m** [comment]
> > @pc-m I don't see the suggestions included in your new commit, was that on purpose?
> 
> No, it's not. I made the changes yesterday messed my rebase this morning

Actually, I think I just skipped this branch :see_no_evil:

**Review by sduthil — APPROVED** [review]

**Pull request merged**